dipy/reconst/shm.py

```diff
@@ -162,21 +163,22 @@ def gen_dirac(m, n, theta, phi, legacy=True):
         `(m + 2) * (m + 1) / 2`.
 
     """
-    return real_sh_descoteaux_from_index(m, n, theta, phi, legacy=legacy)
+    return real_sh_descoteaux_from_index(m_values, l_values, theta, phi,
+                                         legacy=legacy)
 
 
-def spherical_harmonics(m, n, theta, phi, use_scipy=True):
+def spherical_harmonics(m_value, l_value, theta, phi, use_scipy=True):
     """Compute spherical harmonics.
 
     This may take scalar or array arguments. The inputs will be broadcast
     against each other.
 
     Parameters
     ----------
-    m : int ``|m| <= n``
-        The degree of the harmonic.
-    n : int ``>= 0``
-        The order of the harmonic.
+    m_value : int ``|m| <= l``
```

Comment:
I believe that most of the `m_value` and `l_value` in `reconst.shm` are arrays. The old docstring might be erroneous. It might be "array of int" instead of "int".